rust/src/io/exec/take.rs

```diff
@@ -69,8 +69,14 @@ impl Take {
                     let batch = batch?;
                     let row_id_arr = batch.column_by_name(ROW_ID).unwrap();
                     let row_ids: &UInt64Array = as_primitive_array(row_id_arr);
-                    let rows = dataset.take_rows(row_ids.values(), &projection).await?;
-                    let rows = rows.merge(&batch)?;
+                    let rows = if projection.fields.is_empty() {
```

Comment:
How much does an empty take node add to latency im curious

Reply:
let's add some stronger contract between the ExecNode before we try to use empty take node?